app/models/notification_blueprint.rb

```diff
@@ -0,0 +1,24 @@
+# Model to store all information related to a notification.
+# The audience for this notification is calculated on real time by
+# Notification.
+#
+# Foreman and plugins should define NotificationBlueprints for various
+# actions that are notification-worthy. This keeps the responsibilities
+# separate, Notifications taking care of serving the content, and
+# NotificationBlueprint of storing it.
+class NotificationBlueprint < ActiveRecord::Base
+  has_many :notifications
+  belongs_to :subject, :polymorphic => true
+
+  validates :message, :presence => true
+  validates :group, :presence => true
```

Comment:
i would validate name, as you dont want to blueprints with the same name... as a developer you would need to find the blueprint based on its name..